Snowflake.Data.Tests/IntegrationTests/VectorTypesIT.cs

```diff
@@ -0,0 +1,357 @@
+                    reader = (SnowflakeDbDataReader)command.ExecuteReader();
+
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual("[1,2,3]", reader.GetString(0));
+                    var arr = reader.GetArray<int>(0);
+                    Assert.AreEqual(1, arr[0]);
+                    Assert.AreEqual(2, arr[1]);
+                    Assert.AreEqual(3, arr[2]);
+
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual("[4,5,6]", reader.GetString(0));
+                    arr = reader.GetArray<int>(0);
+                    Assert.AreEqual(4, arr[0]);
+                    Assert.AreEqual(5, arr[1]);
+                    Assert.AreEqual(6, arr[2]);
+
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual("[7,8,9]", reader.GetString(0));
+                    arr = reader.GetArray<int>(0);
+                    Assert.AreEqual(7, arr[0]);
+                    Assert.AreEqual(8, arr[1]);
+                    Assert.AreEqual(9, arr[2]);
+
+                    command.CommandText = $"DROP TABLE IF EXISTS {TableName};";
+                    command.ExecuteNonQuery();
+                }
+            }
+        }
+
+        [Test]
+        public void TestSelectFloatVectorFromTable()
+        {
+            using (DbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+                AlterSessionSettings(conn);
+
+                using (DbCommand command = conn.CreateCommand())
+                {
+                    command.CommandText = $"CREATE OR REPLACE TABLE {TableName} (a VECTOR(FLOAT, 3));";
+                    command.ExecuteNonQuery();
+                    command.CommandText = $"INSERT INTO {TableName} SELECT [1.1,2.2,3.3]::VECTOR(FLOAT,3);";
+                    command.ExecuteNonQuery();
+                    command.CommandText = $"INSERT INTO {TableName} SELECT [4.4,5.5,6.6]::VECTOR(FLOAT,3);";
+                    command.ExecuteNonQuery();
+                    command.CommandText = $"INSERT INTO {TableName} SELECT [7.7,8.8,9.9]::VECTOR(FLOAT,3);";
+                    command.ExecuteNonQuery();
+
+                    command.CommandText = $"SELECT COUNT(*) FROM {TableName};";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual(3, reader.GetInt16(0));
+
+                    command.CommandText = $"SELECT * FROM {TableName};";
+                    reader = (SnowflakeDbDataReader)command.ExecuteReader();
+
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual("[1.100000,2.200000,3.300000]", reader.GetString(0));
+                    var arr = reader.GetArray<float>(0);
+                    Assert.AreEqual(1.100000f, arr[0]);
```

Comment:
maybe you could remove the tailing zeros here and below: `Assert.AreEqual(1.f, arr[0]);`

Reply:
Removed